src/Microsoft.ML.TimeSeries/SrCnnEntireAnomalyDetector.cs

```diff
@@ -359,15 +370,45 @@ public void Train(double[] values, ref double[][] results)
                 {
                     Array.Resize<double[]>(ref results, values.Length);
                 }
-                SpectralResidual(values, results, _threshold);
+
+                bool isTemporal = true;
+                double[] seriesToDetect = values.ToArray();
+
+                if (_period > 0)
+                {
+                    StlConfiguration config = new StlConfiguration(_period);
+                    InnerStl stl = new InnerStl(values, config, isTemporal);
```

Comment:
>isTemporal [](start = 64, length = 10)

Why is this needed? Seems to always be true.

Reply:
The decompose can return false in InnerStl::148

---
In reply to: [435221165](https://github.com/dotnet/machinelearning/pull/5202#discussion_r435221165) [](ancestors = 435221165)